src/lundump.c

```diff
@@ -267,6 +270,9 @@ static void loadFunction (LoadState *S, Proto *f, TString *psource) {
   loadUpvalues(S, f);
   loadProtos(S, f);
   loadDebug(S, f);
+  #ifdef USE_YK
+  yk_set_locations(f);
```

Comment:
This is the main fix - set yk locations when `loadFunction` is called. 
Prior to that change, when functions were loaded from a dump, yk locations were not initialised.

Reply:
Nice catch!